### ISSUE TITLE
Refactor Application Gateway http_listener for Consul Ingress 

### ISSUE THREAD

**Issue by ripa1993** [bitrockteam/caravan-infra-azure#21]
Currently blocked by https://feedback.azure.com/forums/217313-networking/suggestions/33841291-reorder-the-listeners-on-the-application-gateway

Once this feature is ready, we can add an http_listener that matches * with lower priority than the consul/nomad/vault ones
